pkg/s3-proxy/server/server.go

```diff
@@ -83,6 +94,14 @@ func (svr *Server) GenerateServer() error {
 		Handler: r,
 	}
 
+	// Get the TLS configuration (if necessary).
+	tlsConfig, err := generateTLSConfig(cfg.Server.SSL, svr.logger)
+	if err != nil {
+		svr.logger.Fatal(err)
```

Comment:
Can you return the error? The main will manage this. I prefer to avoid fatal in pkg as much as possible and let the main handle it.

Reply:
I was following the idiom in this method. Happy to change it. Should I change the existing `svr.logger.Fatal()` (in the OnChangeHook section) as well?

Reply:
The onChangeHook is the special case where I prefer to have it in. The hook should be moved to the main but that requires the generate router function to be public... I've decided to put it in the package for now. Thanks for your local change !

Reply:
Fixed in 004e43f642e5aa047b93146b912338bfb895c423.